File: recursion/mean_list.java
class MeanList{

    public double mean_list(double[] lst, int i, int n){

        if(i==n) return lst[i];
        return lst[i] + mean_list(lst, i+1, n);

    }
    public static void main(String[] args){
        MeanList obj = new MeanList();
        double[] lst = {1,2,3,4,5};
        System.out.println(obj.mean_list(lst, 0, 4)/lst.length);
    }
    
}

File: recursion/mean_list.py
def mean_list(lst, i, n):
    # base case
    if i == n:
        return lst[i]
    # Recursive case
    sum = lst[i] + mean_list(lst, i+1,n)
    # self work
    return sum
    
if __name__ == '__main__':
    lst = [1,2,3,4,5]
    print(mean_list(lst, 0, 4)/len(lst))